builder/builder-manifest.c

```diff
@@ -1735,6 +1735,13 @@ builder_manifest_cleanup (BuilderManifest *self,
       appdata_dir = g_file_resolve_relative_path (app_root, "share/appdata");
       appdata_basename = g_strdup_printf ("%s.appdata.xml", self->id);
       appdata_file = g_file_get_child (appdata_dir, appdata_basename);
+      if (!g_file_test (appdata_file, G_FILE_TEST_EXISTS))
```

Comment:
Oh, i missed this. g_file_test() isn't right, that is an old function that works on pathames, not on GFile. You need to use g_file_query_exists(appdata_file, NULL) instead.